frontend/public/module/k8s/resource.ts

```diff
@@ -103,7 +103,7 @@ export const k8sWatch = (
 
   const path = resourceURL(kind, opts);
   wsOptions.path = path;
-  return new WSFactory(path, wsOptions);
+  return new WSFactory(path, wsOptions as WSOptions);
```

Comment:
Most curious... I guess taking this type up a level (to the input) doesn't work?

Reply:
yeah currently we are receiving only additional params here https://github.com/openshift/console/blob/master/frontend/public/module/k8s/resource.ts#L64-L66

and then mutating it to form the complete options here https://github.com/openshift/console/blob/master/frontend/public/module/k8s/resource.ts#L73-L82

i can avoid mutations and create a new variable to form `wsOptions ` if we really want to avoid `as`

Reply:
Hmmm.... Maybe not to do with your PR. It's not ideal way to do TypeScript types. You're casting and TypeScript is going "unknown object shape... guess you can cast 🤷 "... it's effectively turning off TS.